te_chef.py

```diff
@@ -172,22 +177,52 @@ def scrape_content(title, content_url):
         youtube_url = doc.select_one(".video-container iframe")["src"]
         youtube_id = get_youtube_id_from_url(youtube_url)
 
-        # Some of the videos have been removed from the YouTube channel --
-        # skip creating content nodes for them entirely so they don't show up
-        # as non-loadable videos in Kolibri.
         try:
-            ydl.extract_info(youtube_url, download=False)
+            info = ydl.extract_info(youtube_url, download=False)
+            subtitle_languages = info["subtitles"].keys()
+            print ("      ... with subtitles in languages:", subtitle_languages)
         except youtube_dl.DownloadError as e:
-                print("        NOTE: Skipping video download due to error: ", e)
-                return None
+            # Some of the videos have been removed from the YouTube channel --
+            # skip creating content nodes for them entirely so they don't show up
+            # as non-loadable videos in Kolibri.
+            print("        NOTE: Skipping video download due to error: ", e)
+            return None
 
         video_node = nodes.VideoNode(
             **base_node_attributes,
             derive_thumbnail=True,
             files=[files.YouTubeVideoFile(youtube_id=youtube_id)],
         )
 
-        # TODO(davidhu): Get subtitles in other languages
+        # Add subtitles in whichever languages are available.
+        for language in subtitle_languages:
+            if not languages.getlang(language):
+                # It seems like the subtitle language codes that we get as queried
+                # from youtube-dl are not all consistent with the codes of
+                # the languages in le-utils.
+                #
+                # E.g. we may get "zh-Hans" from
+                # youtube-dl but languages.getlang("zh-Hans") returns None
+                # while languages.getlang("zh") returns something.
+                # Another example, we may get "zu" from youtube-dl but
+                # languages.getlang("zu") returns None while
+                # languages.getlang("zul") returns something ("zul" seems
+                # like the ISO-639-3 version of the language code for Zulu).
```

Comment:
This seems like an inconsistency in `le_utils` since it uses a mix of two-letter codes, two-letter codes with subcode, and three letter codes.

One approach would be to use another library for codes not found by `languages.getlang` and then try to match on name within `le_utils.constants.languages.LANGUAGELIST`. Something like this:

```
from le_utils.constants.languages import LANGUAGELIST
import pycountry

_NAMELOOKUP = {l.name: l for l in LANGUAGELIST}

lang_code = 'zu'
if not languages.getlang(language):

    # try fallback lookup in pycountry
    try:
        pyc_lang = pycountry.languages.get(alpha_2=lang_code)
        language = _NAMELOOKUP.get(pyc_lang.name)
    except KeyError:
        language = None

    if language is None:
        print("      WARNING: subtitle language %s not found in languages"
              " file; skipping download of this subtitle." % lang_code)
        # continue

print(language)  # Language(... primary_code='zul', subcode=None, name='Zulu', ka_name=None)
```